mininet/simpletest.py

```diff
@@ -19,55 +25,94 @@ def pp_json(json_thing, sort=True, indents=4):
         print(json.dumps(json_thing, sort_keys=sort, indent=indents))
 
 
-if __name__ == '__main__':
-    log.setLogLevel('info')
-
-    # Create the network from a given topology without building it yet
-    net = Mininet(topo=SimpleNoBGPTopo(), build=False)
-
-    # Add the controllers to the network
-    c1 = net.addController(name='c1', controller=Floodlight)
-    c2 = net.addController(name='c2', controller=Floodlight)
-
-    # Build the network
-    net.build()
-
+def setUp():
     # Start the switches and specify a controller
     s1 = net.getNodeByName('s1')
-    s1.start([c1])
-    s1.cmd('ovs-vsctl set bridge s1 protocols=OpenFlow15')
     s2 = net.getNodeByName('s2')
-    s2.start([c2])
-    s2.cmd('ovs-vsctl set bridge s2 protocols=OpenFlow15')
 
     # Start each controller
+    c1 = net.getNodeByName('c1')
+    c2 = net.getNodeByName('c2')
     for controller in net.controllers:
         controller.start()
+    s1.start([c1])
+    s2.start([c2])
+
+    # After switch registered, set protocol to OpenFlow 1.5
+    s1.cmd('ovs-vsctl set bridge s1 protocols=OpenFlow15')
+    s2.cmd('ovs-vsctl set bridge s2 protocols=OpenFlow15')
 
     # Send non blocking commands to each host
```

Comment:
We might as well just update this while we're at it. This is just leftover from my stuff